src/atap_widgets/concordance.py

```diff
@@ -134,10 +143,147 @@ def prepare_text_df(
+        elif self.type == "txt":
+            df = self.read_txt()
+            return df
+
+    def read_data_if_chunking(self):
+        """ read_data equivalent for chunking
+        """
+        if self.type == "csv":
+            return pd.read_csv(self.file_location,chunksize = self.chunk)
+        elif self.type == "dataframe":
+            return self.chunk_a_dataframe(self.df_input)
+        elif self.type == "txt":
+            df = self.read_txt()
+            return self.chunk_a_dataframe(df)
+
+    def chunk_a_dataframe(self,df):
+        if df is not None: 
+            list_df = [df[i:i+self.chunk] for i in range(0,df.shape[0],self.chunk)]
+            iter_df = iter(list_df)
+            return iter_df # passthrough input dataframe as an iterable that can be chunked / split
+        else:
+            raise ValueError("type dataframe is enabled, but df_input is None ")
+
+    def read_txt(self):
+        if self.re_symbol_txt is not None: #key symbol value structure
+            
+            b = db.read_text(self.file_location).str.strip().str.split(self.re_symbol_txt)
+            b = b.filter(lambda r : len(r) > 1) 
+            def flatten(record):
+                return {
+                    'key': record[0],
+                    'text': record[1],
+                }
+            df = b.map(flatten).to_dataframe().compute()
+        else: #just read without splitting and assume its all just text - no column structure
+            b = db.read_text(self.file_location).str.strip().filter(lambda r : len(r) > 1) 
+            df = b.to_dataframe().compute()
+            df.columns = ['text']
+            
+        return(df)
+    
+    def process(self):
+        """ Interates over chunks to create a column for grouping
+        """
+        total = pd.DataFrame()
+        for i, df_chunk in enumerate(self.data):
+            new = df_chunk.copy()
+            new['chunk'] = i
+            total = pd.concat([total,new])
+        self.data = total
+        return total
+    
+    def get_grouped_data(self):
+        if text_in_one_doc == False:
+            return self.grouped_data
+        else:
+            return None
+        #return self.group_by_chunk(self.grouped_data) 
+#        return self.grouped_data
+
+#    def get_chunked_data(self):
+#        return self.grouped_data 
+
+    def get_original_data(self):
+        return self.data
+
+    def tag_data(self,df):
+        #add row number tags to original df
+        df['row'] = df.index
+        df['text'] = df['row'].astype(str).str.cat(df['text'],sep = "--") 
+        return df
+
+    def group_by_chunk(self,df):
+        #add row number tags to original df and chunk
+        df['row'] = df.index
+        df['text'] = df['row'].astype(str).str.cat(df['text'],sep = "--") 
+        grouped = df.groupby(['chunk'])['text'].apply(''.join).reset_index()
+        return grouped
+
+    def appify(self,language_model: Union[str, spacy.language.Language] = "en_core_web_sm"):
```

Comment:
What does `appify()` mean here? Can we just call this `show()`?

Reply:
done